pkg/ingester/stream.go

```diff
@@ -594,7 +611,7 @@ func (s *stream) resetCounter() {
 
 func headBlockType(unorderedWrites bool) chunkenc.HeadBlockFmt {
 	if unorderedWrites {
-		return chunkenc.DefaultHeadBlockFmt
+		return chunkenc.UnorderedWithNonIndexedLabelsHeadBlockFmt
```

Comment:
I like the idea. Only problem is once we export such variables outside the `chunkenc`, people use it instead of taking it from `schema` config.

Besides, particularly here this unordered head fmt should also depends on either `chunkv3 or chunkv4` (cannot use normal UnorderedBlockFmt with chunk4 for example). 

I handled it differently by exporting `ChunkHeadFormatFor(chunkfmt)` api from `chunkenc` package.